inginious/frontend/templates/task_dispensers/task_list.html

```diff
@@ -46,11 +46,8 @@
                         </div>
                     </div>
                     <div id="tags_of_task" style="display:none;">
-                        {% for category in task.get_categories() %}
-                            {% set tag = tag_list[category] %}
-                            {% if tag.is_visible_for_student() or user_manager.has_staff_rights_on_course(course) %}
-                                <div id="tag" data-tag-name="{{tag.get_id()}}"></div>
-                            {% endif %}
+                        {% for tag in course.get_task_dispenser().get_categories(task.get_id()) %}
+                            <div id="tag" data-tag-name="{{ tag }}"></div>
```

Comment:
Is this working with everything ? I'm thinking about special chars, maybe it's worth adding some regex, or simply escaping, to ensure it won't break the page.